### PR TITLE
Add notification category deployment and types

### PR DESCRIPTION
**Pull request by shahnami**
This PR adds notification category functionality, sentinel associations and types.

Requires the following to be merged:

- https://github.com/OpenZeppelin/defender-client/pull/191
- https://github.com/OpenZeppelin/defender-client/pull/197

**only merge after a new release of defender-client is published that includes the changes**

### CODE REVIEW

File: examples/defender-test-project/serverless.yml
Comment (pasevin) [inline examples/defender-test-project/serverless.yml:192]: Will need to document this if haven't yet.

**Review by pasevin — APPROVED** [review]
LGTM!

File: examples/defender-test-project/serverless.yml
Comment (mok0230) [inline examples/defender-test-project/serverless.yml:116]: if user has SSOT would it remove the high-severity category? Curious on whether or not we are mandating that these categories exist or if they are optional

File: src/types/docs-schemas/definitions.schema.json
Comment (mok0230) [inline src/types/docs-schemas/definitions.schema.json:169]: can a category be initially created without `notificationIds`?

**Review by mok0230 — APPROVED** [review]
Looks great! Small questions but nothing blocking

File: examples/defender-test-project/serverless.yml
Comment (shahnami) [reply examples/defender-test-project/serverless.yml:116]: Good question! With SSOT enabled, removing these default categories from the YAML, and deploying will also remove it from Defender.

That being said, I don't think we _should_ require these categories to exists (i.e. they can't delete them). The objective should be to have default categories so it's 1 less click, and it incentivises to configure them. What do you think?

File: src/types/docs-schemas/definitions.schema.json
Comment (shahnami) [reply src/types/docs-schemas/definitions.schema.json:169]: Good catch! Yes, that's intentional. The default categories are also created without any notificationIds.

File: examples/defender-test-project/serverless.yml
Comment (mok0230) [reply examples/defender-test-project/serverless.yml:116]: my personal preference would be to allow them to be deleted, but may be good to check with @pasevin that this aligns with the product intent or if these three categories should always exist by the "guide the user into good practices" logic

**Review by BjMrq — APPROVED** [review]

File: examples/defender-test-project/serverless.yml
Comment (shahnami) [reply examples/defender-test-project/serverless.yml:116]: Removed the create and delete capabilities for notification categories.

File: src/cmd/deploy.ts
Comment (mok0230) [inline src/cmd/deploy.ts:601]: Might make sense to throw here and deliver an informative error message (and I suppose maybe this would be part of the client package and fairly certain I just approved the PR). Silly example, but what if they were changing "Low" to "Critical" (because they only have Medium/High/Critical) and changed the notification IDs. If we ignore a name change, we might set their critical channels to low

File: src/utils/index.ts
Comment (mok0230) [inline src/utils/index.ts:268]:
```suggestion
    notificationChannels,
```

**Review by pasevin — APPROVED** [review]

File: src/cmd/deploy.ts
Comment (shahnami) [reply src/cmd/deploy.ts:601]: Not allowing the `name` change is enforced in the API, so not too worried about that. The v0 scope defines only 3 categories, this will change later on, at which point we will enable the create/delete endpoints as well, to allow for more customisation.

As for throwing vs warning, I can see pros and cons to both approaches. Currently we take the `warn` approach for all changes that are not allowed / possible, e.g. sentinel network or type changes. The idea behind it was whether its worth interrupting the entire deploy process for this, what do you think?

Comment (mok0230) [reply src/cmd/deploy.ts:601]: > Not allowing the name change is enforced in the API

Does enforced means it throws or ignores? Where else do we take the approach that an invalid request to update a param ignores? I thought at least in some of those cases we throw

Comment (shahnami) [reply src/cmd/deploy.ts:601]: > Does enforced means it throws or ignores?
At the moment it ignores the `name` field, so essentially only allows to update the description and notification channels for that category.

I don't have a strong opinion on whether we should throw or warn in this case though. Your call.

**Review by pasevin — APPROVED** [review]

File: src/cmd/deploy.ts
Comment (mok0230) [reply src/cmd/deploy.ts:601]: ok, I think changing the name is a bad request given current implementation and should throw (though this should be done on the server side, don't think there are needed changes here) - but would need to be messaged appropriately if users try to do it here

**Review by mok0230 — APPROVED** [review]
good to go once tests pass 🚀

**Pull request merged**
